Mac OS X does not implement unnamed semaphores.  So use a named one instead.

--- osunixxf.c	2010-05-14 17:18:57.000000000 +0200
+++ osunixxf.c	2010-05-14 18:15:57.000000000 +0200
@@ -126,6 +126,7 @@
 #include <sys/time.h>
 #include <semaphore.h>
 #include <pthread.h>
+#include <errno.h>
 
 #include "acpi.h"
 #include "accommon.h"
@@ -537,8 +538,14 @@
 
     if (sem_init (Sem, 0, InitialUnits) == -1)
     {
-        AcpiOsFree (Sem);
-        return (AE_BAD_PARAMETER);
+        if (errno == ENOSYS) {
+            AcpiOsFree (Sem);
+	     Sem = sem_open(tmpnam(NULL), O_EXCL|O_CREAT, 0755, InitialUnits);
+ 	 } else {
+            perror ("AcpiOsCreateSemaphore");
+            AcpiOsFree (Sem);
+            return (AE_BAD_PARAMETER);
+	 }
     }
 
     *OutHandle = (ACPI_HANDLE) Sem;
